node/bft/src/sync/mod.rs

```diff
@@ -436,8 +443,14 @@ impl<N: Network> Sync<N> {
                 trace!("Skipped sending redundant request for certificate {} to '{peer_ip}'", fmt_id(certificate_id));
             }
         }
+        // Calculate the dynamic fetch timeout in milliseconds.
+        let num_validators = self
+            .ledger
+            .get_committee_lookback_for_round(self.storage.current_round())
+            .map_or(Committee::<N>::MAX_COMMITTEE_SIZE as u64, |committee| committee.num_members() as u64);
+        let timeout_in_ms = max_fetch_timeout_in_ms(num_validators);
```

Comment:
since this is the next time this is calculated, maybe a dedicated method would be a good idea?

Reply:
(I can also see this done in the worker module)